src/core_modules/capture-core/events/getSubValues.js

```diff
@@ -37,23 +37,33 @@ const subValueGetterByElementType = {
                 return null;
             }),
     [dataElementTypes.IMAGE]: ({
+        value,
         eventId,
         metaElementId,
         absoluteApiPath,
+        querySingleResource,
     }: {
+        value: any,
         eventId: string,
         metaElementId: string,
         absoluteApiPath: string,
+        querySingleResource: QuerySingleResource,
     }) =>
-        (featureAvailable(FEATURES.trackerImageEndpoint) ?
-            {
-                url: `${absoluteApiPath}/tracker/events/${eventId}/dataValues/${metaElementId}/image`,
-                previewUrl: `${absoluteApiPath}/tracker/events/${eventId}/dataValues/${metaElementId}/image?dimension=small`,
-            } : {
-                url: `${absoluteApiPath}/events/files?dataElementUid=${metaElementId}&eventUid=${eventId}`,
-                previewUrl: `${absoluteApiPath}/events/files?dataElementUid=${metaElementId}&eventUid=${eventId}&dimension=SMALL`,
-            }
-        ),
+        querySingleResource({ resource: `fileResources/${value}` })
```

Comment:
Hi @henrikmv,
IMHO, there is no need to call `/fileResources` because as far as I know, the API for updating events only needs the `value` of DE/TEA and doesn't care about the `name`. We already know the `value`, so something like this should work: 
```
[dataElementTypes.IMAGE]: ({
        value,
        eventId,
        metaElementId,
        absoluteApiPath,
    }: {
        value: any,
        eventId: string,
        metaElementId: string,
        absoluteApiPath: string,
    }) =>
    (featureAvailable(FEATURES.trackerImageEndpoint) ?
            {
                value,
                url: `${absoluteApiPath}/tracker/events/${eventId}/dataValues/${metaElementId}/image`,
                previewUrl: `${absoluteApiPath}/tracker/events/${eventId}/dataValues/${metaElementId}/image?dimension=small`,
            } : {
                value,
                url: `${absoluteApiPath}/events/files?dataElementUid=${metaElementId}&eventUid=${eventId}`,
                previewUrl: `${absoluteApiPath}/events/files?dataElementUid=${metaElementId}&eventUid=${eventId}&dimension=SMALL`,
            }
        ), 
```
Can you re-check? Thank you!

Reply:
Thanks, @simonadomnisoru, that's a good idea, I agree with you!